modules/core/src/lib/api/parse-in-batches.js

```diff
@@ -29,12 +28,9 @@ export async function parseInBatches(data, loaders, options, context) {
     context = null;
   }
 
-  // Extract a url for auto detection
-  const autoUrl = getUrlFromData(data, url);
-
   // Chooses a loader and normalizes it
   // TODO - only uses URL, need a selectLoader variant that peeks at first stream chunk...
-  const loader = selectLoader(null, loaders, options, {url: autoUrl});
+  const loader = selectLoader(data, loaders, options, {url});
```

Comment:
@ibgreen any context I need to know of why `null` was passed here instead of `data`? Also, I noticed that you added `getUrlFromData` recently. This isn't really needed as `selectLoader` calls `getResourceUrlAndType` internally, which does the correct thing.

Reply:
Noticed that `getUrlFromData` is also called for the other `parse` API, while the `data` object gets passed. Will fix that too once I get your feedback here.

Reply:
The problem here is that the type deduction for `parse` includes code that checks the initial bytes of the data. This is problematic with streams and async iterators, since reading the first bytes would consume those bytes from the streams.

As long we can disable the data inspection from streams and async iterators we should be good.

Reply:
`selectLoader` seems to only do data inspection if the loader cannot be found through `url` and `contentType`:

```javascript
  // in select-loader.js
  let loader = findLoaderByUrl(loaders, url || context.url);
  loader = loader || findLoaderByContentType(loaders, type);
  loader = loader || findLoaderByExamingInitialData(loaders, data);
```
What do you propose I do here?

Reply:
> Should we remove the third line?

No that would break things for `parse` users.

Took a quick look it may be fine.

- As long as that function returns null if not passed a string or array buffer we should be good. 
- Equivalently, as long as the extraction of initial bytes from a stream does not happen here.

Reply:
Great. Will merge this PR then.